VirtualListView/VirtualListViewHandler.cs

```diff
@@ -97,36 +97,33 @@ public bool IsItemSelected(int sectionIndex, int itemIndex)
 		return previousSelections.Contains(new ItemPosition(sectionIndex, itemIndex));
 	}
 
-	ItemPosition[] previousSelections = Array.Empty<ItemPosition>();
+	HashSet<ItemPosition> previousSelections = [];
 
 	public static void MapSelectedItems(VirtualListViewHandler handler, IVirtualListView virtualListView)
 	{
 		if (handler is null)
 			return;
 
-		var newSelections = virtualListView?.SelectedItems ?? Array.Empty<ItemPosition>();
+		ISet<ItemPosition> newSelections = virtualListView?.SelectedItems ?? new HashSet<ItemPosition>();
 
 		if (virtualListView.SelectionMode == SelectionMode.None)
-			newSelections = Array.Empty<ItemPosition>();
+			newSelections = new HashSet<ItemPosition>();
 		else if (virtualListView.SelectionMode == SelectionMode.Single && newSelections.Count > 1)
```

Comment:
I see the same problem here

```suggestion
		else if (virtualListView?.SelectionMode == SelectionMode.Single && newSelections.Count > 1)
```